lib/ui/widget/discounts/discount_view_model.dart

```diff
@@ -1,8 +1,52 @@
+import 'package:flutter/material.dart';
 import 'package:stacked/stacked.dart';
+import 'package:flipper/services/flipperNavigation_service.dart';
+import 'package:flipper/services/proxy.dart';
 
 class DiscountViewModel extends BaseViewModel {
-  final String _content = 'New';
+  final FlipperNavigationService _navigationService = ProxyService.nav;
+  void navigateTo({@required String path}) {
+    _navigationService.navigateTo(path);
+  }
+
+  String _content = 'New';
   String get content {
     return _content;
   }
+
+  String _toggle = '%';
+  String get toggle {
+    return _toggle;
+  }
+
+  String _discountAmount = '0.0';
+  String get discountAmount {
+    return _discountAmount;
+  }
+
+  TextEditingController _discount;
```

Comment:
Let's not put TextEditingController in ViewModel anymore! since putting any UI related code in ViewModel make the ViewModel untestable, I added a comment for this in product_description_test class

Reply:
TextEditingController  should stay in view.